### PR TITLE
chore(settings): Convert ProjectTags to FC and useApiQuery

### PR DESCRIPTION
**Pull request by roggenkemper**
this pr converts ProjectTags to be a function component that uses useApiQuery

https://github.com/getsentry/frontend-tsc/issues/2

### CODE REVIEW

File: static/app/views/settings/projectTags/index.tsx [hunk 1]
```diff
@@ -7,140 +7,140 @@ import {Button} from 'sentry/components/button';
...
-        }
+type Props = RouteComponentProps<{projectId: string}, {}>;
+
+type DeleteTagResponse = unknown;
+type DeleteTagVariables = {key: TagWithTopValues['key']};
+
+function ProjectTags(props: Props) {
+  const organization = useOrganization();
+  const {projects} = useProjects();
+  const projectId = props.params.projectId;
```
Comment (evanpurkhiser) [inline static/app/views/settings/projectTags/index.tsx:43]:
```suggestion
  const {projectId} = props.params;
```

File: static/app/views/settings/projectTags/index.tsx [hunk 2]
```diff
@@ -7,140 +7,140 @@ import {Button} from 'sentry/components/button';
...
+
+  const project = useMemo(
+    () => projects.find(p => p.id === projectId),
+    [projects, projectId]
+  );
```
Comment (evanpurkhiser) [inline static/app/views/settings/projectTags/index.tsx:48]: IMO you don't need the memo here. This isn't expensive

File: static/app/views/settings/projectTags/index.tsx [hunk 3]
```diff
@@ -7,140 +7,140 @@ import {Button} from 'sentry/components/button';
...
-    }
-  };
-
-  renderBody() {
-    const {project} = this.props;
-    const {tags} = this.state;
-    const isEmpty = !tags || !tags.length;
+  if (isLoading) {
+    return <LoadingIndicator />;
+  }
```
Comment (evanpurkhiser) [inline static/app/views/settings/projectTags/index.tsx:74]: Should we do error handling too?

@malwilley I think we really should try and build out a helper for doing the loading and error state stuff 😬

File: static/app/views/settings/projectTags/index.tsx [hunk 4]
```diff
@@ -7,140 +7,146 @@ import {Button} from 'sentry/components/button';
...
+    [`/projects/${organization.slug}/${projectId}/tags/`],
+    {staleTime: 0}
+  );
+
+  const {mutate} = useMutation<DeleteTagResponse, RequestError, DeleteTagVariables>({
+    mutationFn: ({key}: DeleteTagVariables) =>
+      api.requestPromise(`/projects/${organization.slug}/${projectId}/tags/${key}/`, {
+        method: 'DELETE',
+      }),
+    onSuccess: (_, {key}) => {
```
Comment (malwilley) [inline static/app/views/settings/projectTags/index.tsx:65]: One more thing to add here is an `onError` that adds an error toast message. I think it's `addErrorMessage('message')`

**Review by malwilley — APPROVED** [review]

**Pull request merged**